include/knowrob/knowrob.h

```diff
@@ -21,6 +21,11 @@ namespace knowrob {
 	 */
 	void InitKnowledgeBase(int argc, char **argv);
 
+	/**
+ 	 * Static initialization of the knowledge base without executable name.
+ 	*/
+	void InitKnowledgeBase();
```

Comment:
I do not think this is safe! Prolog requires argc and argv being correctly initialized to the executable that runs knowrob, here https://github.com/knowrob/knowrob/blob/dev/src/integration/prolog/PrologEngine.cpp#L79

In my experience not setting this will break Prolog

Reply:
I added some wrapper functions for this, that also allow passing the args